android/src/main/kotlin/com/signify/hue/flutterreactiveble/channelhandlers/ScanDevicesHandler.kt

```diff
@@ -49,22 +47,20 @@ class ScanDevicesHandler(private val bleClient: com.signify.hue.flutterreactiveb
         if (this::scanForDevicesDisposable.isInitialized) scanForDevicesDisposable.let {
             if (!it.isDisposed) {
                 it.dispose()
-                scanParameters = null
+                ScanParameterState.setScanParameters(null)
             }
-       }
+        }
     }
 
     fun prepareScan(scanMessage: pb.ScanForDevicesRequest) {
         stopDeviceScan()
         val filter = scanMessage.serviceUuidsList
                 .map { ParcelUuid(UuidConverter().uuidFromByteArray(it.data.toByteArray())) }
         val scanMode = createScanMode(scanMessage.scanMode)
-        scanParameters = ScanParameters(filter, scanMode, scanMessage.requireLocationServicesEnabled)
+        ScanParameterState.setScanParameters(ScanParameters(filter, scanMode, scanMessage.requireLocationServicesEnabled))
```

Comment:
Can't we use an instance variable instead of global state?

Reply:
We need to have global state in a singleton. I can try to put it in a companion object in the eventchannel classt though as last resort but not sure this will fix it.